exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/Fork.java

```diff
@@ -138,13 +150,19 @@ NativeHandle intoPatch() {
 
   /**
    * Creates in-memory checkpoint that can be used to rollback changes.
+   *
+   * <p>Creating a checkpoint will invalidate all collections that were instantiated with this fork.
    */
-  void createCheckpoint() {
+  public void createCheckpoint() {
+    // todo: (1) Are nested rollbacks supported, or is the existing checkpoint lost when a new one
```

Comment:
❗️

Reply:
Nested rollbacks are not possible, only one checkpoint is available at time.

Reply:
OK, I submitted a [task](https://jira.bf.local/browse/ECR-3611) to consider to hide this operation, but I had initially misunderstood how it works and the possible consequences: as multiple rollbacks do not reset the state to the initial fork state, changes made by other services won't be affected.

Reply:
The new test and the updated documentation must hopefully clarify the things.